File: 01_ChangeNext/01_ChangeNext_OC/01_ChangeNext_OC/TitleCell.m
//
//  TitleCell.m
//  01_ChangeNext_OC
//
//  Created by Fengtf on 2016/11/22.
//  Copyright © 2016年 ftf. All rights reserved.
//

#import "TitleCell.h"

@interface TitleCell()


@property (nonatomic,weak)UILabel * titleL;


@end

@implementation TitleCell


- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        UILabel *titleL = [[UILabel alloc]init];
        titleL.textColor = [UIColor blackColor];
        titleL.font = [UIFont systemFontOfSize:12];
        self.titleL = titleL;
        titleL.numberOfLines = 2;
        titleL.textAlignment = NSTextAlignmentLeft;
        [titleL sizeToFit];
        [self addSubview:titleL];
    }
    return self;
}

-(void)layoutSubviews
{
    [super layoutSubviews];

    self.titleL.frame = self.bounds;
}


-(void)setTitle:(NSString *)title
{
    _title = title;

    self.titleL.text = title;
}




@end

File: 01_ChangeNext/01_ChangeNext_Swift/01_ChangeNext_Swift/TitleCell.swift
//
//  TitleCell.swift
//  01_ChangeNext_Swift
//
//  Created by Fengtf on 2016/11/23.
//  Copyright © 2016年 ftf. All rights reserved.
//

import UIKit

class TitleCell: UICollectionViewCell {

    var titleStr:String?{
        didSet{
            titleL.text = titleStr
            titleL.sizeToFit()
        }
    }

    let titleL = UILabel()

    override init(frame: CGRect) {
        super.init(frame: frame)

        self .addSubview(titleL)
        titleL.textColor = UIColor.black
        titleL.font = UIFont.systemFont(ofSize: 12)
        titleL.numberOfLines = 2
        titleL.textAlignment = .left
        titleL.sizeToFit()
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        titleL.frame = self.bounds
    }

    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }



}
